### PR TITLE
Add Reminder Command For Users to View Reminders

### PR DESCRIPTION
**Pull request by DanielLimWeiEn**
## Reminder Command

Added the logic to Friendnancial for it to handle the `remind` command. The `remind` command is typed as is and it creates a popup window that displays all the names and birthdays of every person as reminders to the user.

Logic and Parser Components:
* `CommandResult` - Modified, to handle `ReminderCommand` results.
* `AddressBookParser`- Modified, to handle the parsing the `remind` command from the user.
* `ReminderCommand`- Added, to handle `ReminderCommand`s.

UI Components:
* `MainWindow` - Modified, to handle displaying the `ReminderWindow` component on the GUI.
* `ReminderWindow` - Added, the component used to display reminders on the GUI. Inspiration taken from `MainWindow` (displaying a list of Persons) and `HelpWindow` (popup window).
* `ReminderListPanel` - Added, based off of `PersonListPanel`, to hold the list of reminders.
* `ReminderCard` - Added, based off of `PersonCard`, a card for holding the reminders.
* `ReminderWindow.fxml` - Added, corresponds to what to display on the GUI for `ReminderWindow` component.
* `ReminderListPanel.fxml` - Added, corresponds to what to display on the GUI for `ReminderListPanel` component.
* `ReminderListCard.fxml` - Added, corresponds to what to display on the GUI for `ReminderCard` component.
* `ReminderWindow.css` - Modified, change styling.

Testing:
* `CommandResultTest` - Modified, to handle testing the behaviour of the updated `CommandResult` objects.
* `AddressBookParserTest` - Modified, to handle testing the parsing of the `remind` command.
* `ReminderCommandTest` - Added, to handle testing of the `ReminderCommand` objects. Inspiration taken from `EditCommandTest`.

Not Done:
- [ ] Updating Docs with Pictures
- [ ] Adding automated GUI testing using TestFX for the `HelpCommand` and subsequently, the `RemindCommand`.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/CommandResult.java [hunk 1]
```diff
@@ -17,13 +17,29 @@ public class CommandResult {
     /** The application should exit. */
     private final boolean exit;
 
+    /** Reminder information should be shown to the user. */
+    private final boolean showReminder;
```
Comment (tyw2811) [inline src/main/java/seedu/address/logic/commands/CommandResult.java:21]: Perhaps a more intuitive boolean name can be used here

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 2]
```diff
@@ -163,6 +166,18 @@ private void handleExit() {
...
+     * Opens the reminder window or focuses on it if it's already opened.
+     */
+    @FXML
+    public void handleReminder() {
+        if (!reminderWindow.isShowing()) {
+            reminderWindow.show();
+        } else {
+            reminderWindow.focus();
+        }
+    }
```
Comment (tyw2811) [inline src/main/java/seedu/address/ui/MainWindow.java:179]: I like how the case of window being opened already is handled.

**Review by tyw2811 — COMMENTED** [review]
LGTM!

File: src/main/java/seedu/address/logic/commands/CommandResult.java (hunk 1 above)
Comment (DanielLimWeiEn) [reply src/main/java/seedu/address/logic/commands/CommandResult.java:21]: The intent of the variable is to mark whether to show reminder window or not similar to help window. Hence I think that it signals the intent sufficiently

**Pull request merged**